connection.go

```diff
@@ -1979,7 +1977,7 @@ func (s *connection) onStreamCompleted(id protocol.StreamID) {
 }
 
 func (s *connection) SendMessage(p []byte) error {
-	if s.datagramQueue == nil {
+	if s.datagramQueue == nil || !s.supportsDatagrams() {
```

Comment:
I think you can remove the `s.datagramQueue == nil` check. You initialize the `datagramQueue` in all cases.